pkg/authorization/reaper/cluster_role.go

```diff
@@ -0,0 +1,52 @@
+package reaper
+
+import (
+	"time"
+
+	"github.com/golang/glog"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kerrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/kubectl"
+	"k8s.io/kubernetes/pkg/labels"
+
+	"github.com/openshift/origin/pkg/client"
+)
+
+func NewClusterRoleReaper(roleClient client.ClusterRolesInterface, bindingClient client.ClusterRoleBindingsInterface) kubectl.Reaper {
+	return &ClusterRoleReaper{
+		roleClient:    roleClient,
+		bindingClient: bindingClient,
+	}
+}
+
+type ClusterRoleReaper struct {
+	roleClient    client.ClusterRolesInterface
+	bindingClient client.ClusterRoleBindingsInterface
+}
+
+// Stop on a reaper is actually used for deletion.  In this case, we'll delete referencing clusterrolebindings
+// then delete the clusterrole.
+func (r *ClusterRoleReaper) Stop(namespace, name string, timeout time.Duration, gracePeriod *kapi.DeleteOptions) (string, error) {
```

Comment:
Noting two changes @deads2k and I want:
* `oc replace --force` should only delete if a Conflict error is received
* `oc replace --force` should default --cascade to `false`